alarm/nialarm.yml

```diff
@@ -0,0 +1,790 @@
+    url: 'https://www.ni.com/systemlink'
+    email: support@ni.com
+basePath: /nialarm
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  basicAuth:
+    type: basic
+security:
+  - basicAuth: []
+x-ni-routing-key: Skyline.AlarmInstance
+definitions:
+  Error:
+    description: Contains error information
+    type: object
+    properties:
+      name:
+        description: String error code
+        type: string
+      code:
+        description: Numeric error code
+        type: integer
+      message:
+        description: Complete error message
+        type: string
+      args:
+        description: Positional argument values for the error code
+        type: array
+        items:
+          type: string
+    example:
+      name: AlarmInstance.InstanceNotFound
+      code: -253006
+      message: An instance with instanceId '5c2cf7e0e0d64403b486fcb4' was not found.
+      args: []
+  Operation:
+    description: Operation provided by the API
+    type: object
+    properties:
+      available:
+        type: boolean
+        description: Whether the operation is available to the caller
+      version:
+        type: integer
+        description: Version of the available operation
+    required: [available]
+    example:
+      available: true
+      version: 1
+  V1Operations:
+    title: V1 Operations
+    description: V1 operations
+    type: object
+    properties:
+      operations:
+        description: >-
+          Available operations in the v1 version of the API:
+
+          - deleteAlarmInstances: The ability to delete alarm instances
+
+          - readAlarmInstances: The ability to query for alarm instances
+
+          - writeAlarmInstances: The ability to create and modify alarm instances
+
+        type: object
+        properties:
+          deleteAlarmInstances:
+            $ref: '#/definitions/Operation'
+          readAlarmInstances:
+            $ref: '#/definitions/Operation'
+          writeAlarmInstances:
+            $ref: '#/definitions/Operation'
+  AlarmInstance:
+    title: Alarm Instance
+    description: An individual instance of an alarm. The lifecycle of an instance of an alarm begins the first
+      time it is triggered and ends when it has been both acknowledged and cleared. The services enforces the
+      invariant that there can be at most one *active*/true instance of an alarm with a given *alarmId*.
+    type: object
+    required:
+      - alarmId
+    properties:
+      instanceId:
+        description: The unique ID of a particular instance of an alarm.
+        type: string
+        example: 5c33c212e0d6444320d9a9f4
+      alarmId:
+        description: A value meant to uniquely identify the particular process or condition tracked by an
+          alarm. For example, alarm instances created by the Tag Rule Engine Service have their *alarmIds* set
+          to the concatenation of the path of the tag which caused the rule to be triggered and the ID of the
```

Comment:
So it isn't really the tag's path which caused the rule to be triggered, it's the fact that the tag's value is within the set point. The path of the tag uniquely identifies the tag, so that's what we append to the alarmId.